### PR TITLE
Update stats analytics

### LINKED ISSUE #20176: Add or update analytics
See https://github.com/wordpress-mobile/jetpack-issue-repo/issues/4

The current events will be updated based on the discussion at https://github.com/wordpress-mobile/jetpack-issue-repo/issues/4.

### PR DESCRIPTION
**Pull request by irfano**
Fixes #20176 

This updates analytics events related to the stats screen.

|Event|Description|
|-|-|
|`stats_insights_accessed`|[FIXED] After opening the stats screen, this was always being tracked; even the insights tab was not selected. This fixes it and tracks the correct event.|
|`stats_period_accessed`|will be tracked when the TRAFFIC tab is shown. This event was including both `granularity` and `period` properties with the same value. Now it will include only "period".|
|`stats_previous_date_tapped`|`stats_previous_date_tapped` renamed to `stats_date_tapped_backward` to match with iOS. Before it was sending null period, it has been fixed.|
|`stats_next_date_tapped`|`stats_next_date_tapped` renamed to `stats_date_tapped_forward` to match with iOS. Before it was sending null period, it has been fixed.|

|`tap_source` property that has been added to `stats_accesed` event|Description|
|-|-|
|`quick_actions`|opening stats from My Site|
|`row`|opening stats from My Site → More|
|`today_stats_card`|opening stats from "Today's Stats" card on "My Site"|
|`notification`|opening stats from a notification|
|`link`|opening stats from a deeplink|
|`shortcut`|opening stats from shortcut|
|`activity_log`|opening stats from Activity Log screen|
|`jetpack_connection`|opening stats after connecting Jetpack|
|`widget`|opening stats from widgets|

-----

## To Test:
Test these for both disabled and enabled `stats_traffic_tab` flag. The flag can be set from "Me → Debug Settings".

`stats_insights_accessed`
1. Log in.
2. Open stats from My Site.
3. If the INSIGTHS tab is selected, verify that the `🔵 Tracked: stats_insights_accessed` is logged.
4. Change the tab and relesect INSIGHTS tab.
5. Verify that the `🔵 Tracked: stats_insights_accessed` is logged again.

`stats_period_accessed`
1. Log in.
2. Ensure the `stats_traffic_tab` is disabled.
3. Open stats from My Site.
4. Verify that when DAYS tab is selected, `🔵 Tracked: stats_period_accessed, Properties: {"period":"days"}` is logged. 
5. Select other granular tabs and verify that period property is updated properly.
6. Enable `stats_traffic_tab` flag.
7. Open stats from My Site.
8. Verify that when TRAFFIC tab is selected, `🔵 Tracked: stats_period_accessed, Properties: {"period":"days"}` is logged. 

`stats_previous_date_tapped`, `stats_next_date_tapped`
1. Log in.
2. Ensure the flag is disabled.
3. Open stats from My Site.
4. Select DAYS, WEEKS, MONTHS or YEARS tab.
5. Change the selected date to the previous date with arrows.
6. Verify that `🔵 Tracked: stats_date_tapped_backward, Properties: {"period":"days"}` is logged with the proper period.
7. Change the selected granularity and repeat 6.
8. Change the selected date to the next date with arrows.
9. Verify that `🔵 Tracked: stats_date_tapped_forward, Properties: {"period":"days"}` is logged with the proper period.
10. Change the selected granularity and repeat 9.
**11. Disable the feature flag.**
12. Open stats from My Site.
13. Select the TRAFFIC tab.
14. Repeat 5-6-7-8-9-10.

`quick_actions`
1. Log in.
2. Open stats from My Site.
3. Verify that `🔵 Tracked: stats_accessed, Properties: {"tap_source":"quick_actions",...}` is logged.

`row`
1. Log in.
2. Open stats from My Site → More.
3. Verify that `🔵 Tracked: stats_accessed, Properties: {"tap_source":"row",...}` is logged.

`today_stats_card`
1. Log in.
2. Open stats from "Today's Stats" card on "My Site".
3. Verify that `🔵 Tracked: stats_accessed, Properties: {"tap_source":"today_stats_card",...}` is logged.

`notification`
_You need to use debug build to test this or wait Monday for weekly roundup notification._
1. Log in.
2. Ensure notification permission is allowed.
3. Open Me → Debug settings.
4. Tap the menu button at the top right.
5. Tap "Force show Weekly Roundup notification" button.
6. Wait for a while, ~30 seconds.
7. Tap the Weekly Roundup notification received.
8. Verify that `🔵 Tracked: stats_accessed, Properties: {"tap_source":"notification",...}` is logged.

`link`
_If you know an easier way to test a deeplink, you can use it._
1. Log in.
2. Open internet browser on the device and open any online HTML editor that you can compile the HTML code.
3. Type this code into HTML editor:
```
<html>
<p>Test <a href="jetpack://stats">jetpack://stats</a> deeplink</p>
</html>
```
4. After running the code, tap the `jetpack://stats` link.
5. Verify that the stats screen is opened and `🔵 Tracked: stats_accessed, Properties: {"tap_source":"link",...}` is logged.

`shortcut`
1. Log in.
2. Open the device's home screen.
3. Long tap on the app icon.
4. From the shortcut menu, tap Stats.
5. Verify that the stats screen is opened and `🔵 Tracked: stats_accessed, Properties: {"tap_source":"shortcut",...}` is logged.

`jetpack_connection`
1. Log in with a self-hosted site that is not connected to the Jetpack.
2. Open Stats from any point, e.g. "My Site →Stats".
3. You will see the INSTALL JETPACK screen instead of the stats.
4. Tap INSTALL JETPACK and complete the flow.
5. The stats screen should be shown at the end of the flow.
6. Verify that `🔵 Tracked: stats_accessed, Properties: {"tap_source":"jetpack_connection",...}` is logged.

`activity_log`
I couldn't find a way to test this. Please let me know if you can test it. I'm also pretty sure this PR will track the `activity_log` property properly.

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

16. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

17. What automated tests I added (or what prevented me from doing so)

    - Updated current unit tests

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist:

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by ravishanker** [comment]
> stats_date_tapped_backward **renamed** to stats_previous_date_tapped to match with iOS.
> stats_date_tapped_forward **renamed** to stats_next_date_tapped  to match with iOS

It is still showing...backward, and forward, instead of previous_date, and next_date.  Test instructions also differ from what is stated in Description about renaming.

🔵 Tracked: stats_date_tapped_backward, Properties: {"period":"days"}
🔵 Tracked: stats_date_tapped_forward, Properties: {"period":"weeks"}

**Comment by ravishanker** [comment]
> > stats_date_tapped_backward **renamed** to stats_previous_date_tapped to match with iOS.
> > stats_date_tapped_forward **renamed** to stats_next_date_tapped  to match with iOS
> 
> It is still showing...backward, and forward, instead of previous_date, and next_date. Test instructions also differ from what is stated in Description about renaming.
> 
> 🔵 Tracked: stats_date_tapped_backward, Properties: {"period":"days"} 🔵 Tracked: stats_date_tapped_forward, Properties: {"period":"weeks"}

Actually, the description is wrong implementation d4231a1c73b7b4b1289556e112cbd0f94845e5f7 is correct!

**Review by ravishanker — APPROVED** [review]
LGTM 👍

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/277">24.3</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by irfano** [comment]
> Actually, the description is wrong implementation https://github.com/wordpress-mobile/WordPress-Android/commit/d4231a1c73b7b4b1289556e112cbd0f94845e5f7 is correct!

Thank you for noticing. I have corrected the description.

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/utils/FormattableContentClickHandler.kt [hunk 1]
```diff
@@ -117,7 +118,7 @@ class FormattableContentClickHandler @Inject constructor(
         if (rangeType == FormattableRangeType.FOLLOW) {
             ActivityLauncher.viewAllTabbedInsightsStats(activity, FOLLOWERS, 0, site.id)
         } else {
-            ActivityLauncher.viewBlogStats(activity, site)
+            ActivityLauncher.viewBlogStats(activity, site, StatsActivity.StatsLaunchedFrom.ACTIVITY_LOG)
```
Comment (staskus) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/utils/FormattableContentClickHandler.kt:121]: Is this method related to `activity_log`? If I read the code correctly this is about opening Stats from Notification details view, if yes it should be `notification` rather than `activity_log`. @irfano

Comment (irfano) [reply WordPress/src/main/java/org/wordpress/android/ui/notifications/utils/FormattableContentClickHandler.kt:121]: Thanks for reviewing, @staskus! I thought the same at first. But actually, `FormattableContentClickHandler` is being used only in [ActivityLogDetailFragment](https://github.com/wordpress-mobile/WordPress-Android/blob/380a6de79ca90947814a3c7e87916a71329e0442/WordPress/src/main/java/org/wordpress/android/ui/activitylog/detail/ActivityLogDetailFragment.kt#L152-L156).

Comment (staskus) [reply WordPress/src/main/java/org/wordpress/android/ui/notifications/utils/FormattableContentClickHandler.kt:121]: Thanks! It gave me more pointers to look for similar code on iOS as well 👍
